src/Microsoft.Kusto.ServiceLayer/DataSource/KustoResultsReader.cs

```diff
@@ -10,13 +12,15 @@ public KustoResultsReader(IDataReader reader)
         }
 
         /// <summary>
-        /// Kusto returns 3 results tables - QueryResults, QueryProperties, QueryStatus. When returning query results
-        /// we want the caller to only read the first table. We override the NextResult function here to only return one table
-        /// from the IDataReader.
+        /// Kusto returns atleast 4 results tables - QueryResults(sometimes more than one), QueryProperties, QueryStatus and Query Results Metadata Table. 
+        /// When returning query results we need to trim off the last 3 tables as we want the caller to only read results table. 
         /// </summary>
-        /*public override bool NextResult()
+        public void SanitizeResults(List<ResultSet> resultSets)
         {
-            return false;
-        }*/
+            if (resultSets.Count > 3)
```

Comment:
Can't this entire logic be contained in this file. The result set can be trimmed in the constructor or in NextResults. Without any changes needed in QueryExecution. This whole class was created so that we could override the results and return as needed, without changing the rest of the code.

Reply:
We need changes to QueryExecution as the logic there send the tables to ADS as it reads. We want to read all the results and trim them out before sending the results to ADS.